File: spsc_web/WebContent/samples.js
var programs = new Array();

programs["appendXYaZ"]=
'append(Nil, vs) = vs;\n\
append(Cons(u, us), vs) = Cons(u, append(us, vs));\n\
appendXYaZ(xs, ys, zs) = append(append(xs, ys), zs);'

programs["appendXYaX"]=
"append(Nil, vs) = vs;\n\
append(Cons(u, us), vs) = Cons(u, append(us, vs));\n\
appendXYaX(xs, ys) = append(append(xs, ys), xs);"

programs["appendXaYZ"]=
"append(Nil, vs) = vs;\n\
append(Cons(u, us), vs) = Cons(u, append(us, vs));\n\
appendXaYZ(xs, ys, zs) = append(xs, append(ys, zs));"

programs["takenm"]=
"from(n) = Cons(n, from(S(n)));\n\
head(Cons(x, xs)) = x;\n\
mapAdd1(Nil) = Nil;\n\
mapAdd1(Cons(x, xs)) = Cons(S(x), mapAdd1(xs));\n\
tail(Cons(x, xs)) = xs;\n\
take(Z, xs) = Nil;\n\
take(S(n), xs) = Cons(head(xs), take(n, tail(xs)));\n\
takenm(n, m) = take(n, mapAdd1(from(m)));"

programs["mapfrom"]=
"from(n) = Cons(n, from(S(n)));\n\
mapAdd1(Nil) = Nil;\n\
mapAdd1(Cons(x, xs)) = Cons(S(x), mapAdd1(xs));\n\
mapfrom(n) = mapAdd1(from(n));"

programs["abab"]=
"ab(A(x)) = B(ab(x));\n\
ab(B(x)) = A(ab(x));\n\
abab(x) = ab(ab(x));"

programs["reverse"]=
"app(Nil, vs) = vs;\n\
app(Cons(u, us), vs) = Cons(u, app(us, vs));\n\
rev(Nil) = Nil;\n\
rev(Cons(x, xs)) = app(rev(xs), Cons(x, Nil));\n\
reverse(x)=rev(x);"

programs["eqxx"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
eqxx(x) = eq(x, x);"

programs["eqxSx"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
eqxSx(x) = eq(x, S(x));"

programs["eqSxx"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
eqSxx(x) = eq(S(x), x);"

programs["eqSZx"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
eqSZx(x) = eq(S(Z), x);"

programs["mapNotNot"]=
"not(True) = False;\n\
not(False) = True;\n\
mapNot(Nil) = Nil;\n\
mapNot(Cons(x, xs)) = Cons(not(x), mapNot(xs));\n\
\n\
mapNotNot(xs) = mapNot(mapNot(xs));"

programs["mapNotNotNot"]=
"not(True) = False;\n\
not(False) = True;\n\
mapNot(Nil) = Nil;\n\
mapNot(Cons(x, xs)) = Cons(not(x), mapNot(xs));\n\
\n\
mapNotNotNot(xs) = mapNot(mapNot(mapNot(xs)));"

programs["member"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
\n\
memberSel(Nil, x) = False;\n\
memberSel(Cons(y, ys), x) = memberEq(eq(x, y), x, ys);\n\
\n\
memberEq(True, x, ys) = True;\n\
memberEq(False, x, ys) = memberSel(ys, x);\n\
\n\
member(x, ys) = memberSel(ys, x);"

programs["memberSZ"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
\n\
memberSel(Nil, x) = False;\n\
memberSel(Cons(y, ys), x) = memberEq(eq(x, y), x, ys);\n\
\n\
memberEq(True, x, ys) = True;\n\
memberEq(False, x, ys) = memberSel(ys, x);\n\
\n\
member(x, ys) = memberSel(ys, x);\n\
memberSZ(list) = member(S(Z), list);"

programs["memberZSZ"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
\n\
memberSel(Nil, x) = False;\n\
memberSel(Cons(y, ys), x) = memberEq(eq(x, y), x, ys);\n\
\n\
memberEq(True, x, ys) = True;\n\
memberEq(False, x, ys) = memberSel(ys, x);\n\
\n\
member(x, ys) = memberSel(ys, x);\n\
memberZSZ(x) = member(x, Cons(Z, Cons(S(Z), Nil)));"

programs["memberLazy"]=
"eq(Z, y) = eqZ(y);\n\
eq(S(x), y) = eqS(y, x);\n\
eqZ(Z) = True;\n\
eqZ(S(x)) = False;\n\
eqS(Z, x) = False;\n\
eqS(S(y), x) = eq(x, y);\n\
\n\
null(Nil) = True;\n\
null(Cons(x, xs)) = False;\n\
\n\
head(Cons(x, xs)) = x;\n\
tail(Cons(x, xs)) = xs;\n\
\n\
if(True, x, y) = x;\n\
if(False, x, y) = y;\n\
\n\
not(x) = if(x, False, True);\n\
or(x, y) = if(x, True, y);\n\
and(x, y) = if(x, y, False);\n\
\n\
memberLazy(x, list) = and(not(null(list)), or(eq(x, head(list)), memberLazy(x, tail(list))));"

programs["notornot"]=
"if(True, x, y) = x;\n\
if(False, x, y) = y;\n\
not(x) = if(x, False, True);\n\
or(x, y) = if(x, True, y);\n\
and(x, y) = if(x, y, False);\n\
notornot(x, y) = not(or(not(x), not(y)));"

function sample(sampleName){
  var fname = document.getElementById('fname');
  fname.value = sampleName;
  var programText = document.getElementById('programText');
  programText.value = programs[sampleName];
  return false;	
}

File: spsc-scala/spsc/src/main/scala/spsc/Samples.scala
package spsc

object Samples extends App {
  SuperCompilerApp.main(Array(
    "-i", "samples/success/append.sl",
    "-f", "appendXYaZ",
    "-t", "samples/success/appendXYaZ.out.svg",
    "-p", "samples/success/appendXYaZ.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/append.sl",
    "-f", "appendXYaX",
    "-t", "samples/success/appendXYaX.out.svg",
    "-p", "samples/success/appendXYaX.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/append.sl",
    "-f", "appendXaYZ",
    "-t", "samples/success/appendXaYZ.out.svg",
    "-p", "samples/success/appendXaYZ.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/fail/takenm.sl",
    "-f", "takenm",
    "-t", "samples/fail/takenm.out.svg",
    "-p", "samples/fail/takenm.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/fail/mapfrom.sl",
    "-f", "mapfrom",
    "-t", "samples/fail/mapfrom.out.svg",
    "-p", "samples/fail/mapfrom.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/abab.sl",
    "-f", "abab",
    "-t", "samples/success/abab.out.svg",
    "-p", "samples/success/abab.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/fail/reverse.sl",
    "-f", "reverse",
    "-t", "samples/fail/reverse.out.svg",
    "-p", "samples/fail/reverse.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/eq.sl",
    "-f", "eqxx",
    "-t", "samples/success/eqxx.out.svg",
    "-p", "samples/success/eqxx.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/eq.sl",
    "-f", "eqxSx",
    "-t", "samples/success/eqxSx.out.svg",
    "-p", "samples/success/eqxSx.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/eq.sl",
    "-f", "eqSxx",
    "-t", "samples/success/eqSxx.out.svg",
    "-p", "samples/success/eqSxx.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/eq.sl",
    "-f", "eqSZx",
    "-t", "samples/success/eqSZx.out.svg",
    "-p", "samples/success/eqSZx.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/mapnot.sl",
    "-f", "mapNotNot",
    "-t", "samples/success/mapNotNot.out.svg",
    "-p", "samples/success/mapNotNot.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/mapnot.sl",
    "-f", "mapNotNotNot",
    "-t", "samples/success/mapNotNotNot.out.svg",
    "-p", "samples/success/mapNotNotNot.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/member.sl",
    "-f", "member",
    "-t", "samples/success/member.out.svg",
    "-p", "samples/success/member.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/member.sl",
    "-f", "memberSZ",
    "-t", "samples/success/memberSZ.out.svg",
    "-p", "samples/success/memberSZ.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/member.sl",
    "-f", "memberZSZ",
    "-t", "samples/success/memberZSZ.out.svg",
    "-p", "samples/success/memberZSZ.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/fail/memberLazy.sl",
    "-f", "member",
    "-t", "samples/fail/memberLazy.out.svg",
    "-p", "samples/fail/memberLazy.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/notornot.sl",
    "-f", "notornot",
    "-t", "samples/success/notornot.out.svg",
    "-p", "samples/success/notornot.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/flip.sl",
    "-f", "flip2",
    "-t", "samples/success/flip2.out.svg",
    "-p", "samples/success/flip2.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/flip.sl",
    "-f", "flip3",
    "-t", "samples/success/flip3.out.svg",
    "-p", "samples/success/flip3.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/success/add.sl",
    "-f", "e",
    "-t", "samples/success/add.out.svg",
    "-p", "samples/success/add.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/input/kmpNestedCalls.sl",
    "-f", "matchAAB",
    "-t", "samples/input/kmpNestedCalls.out.svg",
    "-p", "samples/input/kmpNestedCalls.out.sl"
  ))

  SuperCompilerApp.main(Array(
    "-i", "samples/input/kmpTailCalls.sl",
    "-f", "matchAAB",
    "-t", "samples/input/kmpTailCalls.out.svg",
    "-p", "samples/input/kmpTailCalls.out.sl"
  ))
}
